src/opera/api/controllers/background_invocation.py

```diff
@@ -71,8 +72,12 @@ def run_internal(work_queue: multiprocessing.Queue):
             except BaseException as e:
                 if isinstance(e, RuntimeError):
                     raise e
+                elif isinstance(e, ParseError):
+                    inv.exception = "{}: {}: {}\n\n{}".format(e.__class__.__name__, e.loc, str(e),
+                                                              traceback.format_exc())
+                else:
+                    inv.exception = "{}: {}\n\n{}".format(e.__class__.__name__, str(e), traceback.format_exc())
                 inv.state = InvocationState.FAILED
```

Comment:
```suggestion
            except RuntimeError as e:
                inv.state = InvocationState.FAILED
                raise e
            except ParseError as e:
                inv.state = InvocationState.FAILED
                inv.exception = "{}: {}: {}\n\n{}".format(e.__class__.__name__, e.loc, str(e),
                                                          traceback.format_exc())
            except BaseException as e:
                inv.state = InvocationState.FAILED
                inv.exception = "{}: {}\n\n{}".format(e.__class__.__name__, str(e), traceback.format_exc())
                
```

Reply:
I would propose to do it in more coherent way.
By the way in your case if RuntimeError exception is caught InvocationState is not set in FAILED way. Was it by design?

Reply:
I agree with your coherent approach. I also agree RuntimeError exception handling is not the best in our context, it was adapted after [xopera-api](https://github.com/xlab-si/xopera-api) and probably makes more sense there in single-user-single-project environment. However, your suggestion does not fully fix it, invocation object must be saved to DB in order for a user to be able to read it. I will introduce a patch for it.